src/static/index.js

```diff
@@ -225,20 +240,18 @@ $.getJSON('/user-config/config.json', (result) => {
     },
   });
 
-  if (sessionManager.isUserLoggedIn) {
+  if (sessionManager.isLoggedIn) {
     astarteClient.setCredentials(sessionManager.getCredentials());
   }
 
-  // TODO use TargetEvent API or custom
-  // events library
-  sessionManager.onSessionChange = (newSession) => {
-    app.ports.onSessionChange.send(newSession);
+  sessionManager.on('sessionChange', (newSession) => {
+    app.ports.onSessionChange.send(JSON.parse(SessionManager.serializeSession(newSession)));
```

Comment:
why do you serialize and parse it on the very same lane? can we just pass it to elm?

Reply:
The only trick here is that Elm still needs a session and I preferred not to touch the Elm parsing/implementation.
So on sessionChange, I am serializing the session into the format Elm understands; the `JSON.parse` step is there just because the Elm port expects an object, not a string